src/placeholder/placeholder.service.ts

```diff
@@ -10,59 +10,89 @@ import { Injectable } from "@angular/core";
  * Singleton service used to register the container for out-of-flow components to insert into.
  * Also used to insert/remove components from that view.
  */
-@Injectable()
+@Injectable({ providedIn: "root" })
 export class PlaceholderService {
 	/**
-	 * Maintain a `ViewContainerRef` to an instance of the component.
+	 * Main `ViewContainerRef` to insert components into
 	 */
 	protected viewContainerRef: ViewContainerRef = null;
+	/**
+	 * Map of id's to secondary `ViewContainerRef`s
+	 */
+	protected viewContainerMap: Map<any, ViewContainerRef> = new Map();
 	/**
 	 * Used by `Placeholder` to register view-container reference.
 	 */
-	registerViewContainerRef(vcRef: ViewContainerRef): void {
-		this.viewContainerRef = vcRef;
+	registerViewContainerRef(vcRef: ViewContainerRef, id?: any): void {
+		if (id) {
+			this.viewContainerMap.set(id, vcRef);
+		} else {
+			this.viewContainerRef = vcRef;
+		}
 	}
 
 	/**
 	 * Creates and returns component in the view.
 	 */
-	createComponent(componentFactory: ComponentFactory<any>, injector: Injector): ComponentRef<any> {
+	createComponent(componentFactory: ComponentFactory<any>, injector: Injector, id?: any): ComponentRef<any> {
+		if (id) {
+			if (!this.viewContainerMap.has(id)) {
+				console.error(`No view container with id ${id} found`);
+				return;
+			}
+			return this.viewContainerMap.get(id).createComponent(componentFactory, null, injector);
+		}
 		if (!this.viewContainerRef) {
 			console.error("No view container defined! Likely due to a missing `ibm-placeholder`");
+			return;
 		}
 		return this.viewContainerRef.createComponent(componentFactory, null, injector);
 	}
 
 	destroyComponent(component: ComponentRef<any>) {
-		const index = this.viewContainerRef.indexOf(component.hostView);
-		if (index < 0) {
-			return;
-		}
+		component.destroy();
+		// const index = this.viewContainerRef.indexOf(component.hostView);
+		// if (index < 0) {
+		// 	return;
+		// }
 
-		// destroy the view
-		this.viewContainerRef.remove(index);
+		// // destroy the view
+		// this.viewContainerRef.remove(index);
```

Comment:
?

Reply:
Aw shoot, good catch ... definitely not supposed to be there